core-addon/DataCollection.js

```diff
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = class DataCollection {
+  /**
+   * Sends an empty Ion ping with the provided info.
+   *
+   * @param {String} payloadType
+   *        The type of the encrypted payload. This will define the
+   *        `schemaName` of the ping.
+   *
+   * @param {String} namespace
+   *        The namespace to route the ping. This will define the
+   *        `schemaNamespace` and `studyName` properties of the ping.
+   */
+  async _sendEmptyPing(payloadType, namespace) {
+    // NOTE - while we're not actually sending useful data in
+    // this payload, the current Ion v2 Telemetry pipeline requires
+    // that pings are shaped this way so they are routed to the correct
+    // environment.
+    //
+    // At the moment, the public key used here isn't important but we do
+    // need to use *something*.
+    const publicKey = {
+      crv: "P-256",
+      kty: "EC",
+      x: "XLkI3NaY3-AF2nRMspC63BT1u0Y3moXYSfss7VuQ0mk",
+      y: "SB0KnIW-pqk85OIEYZenoNkEyOOp5GeWQhS1KeRtEUE",
+    };
+
+    await this.sendPing(
+      payloadType,
+      // We expect to send an empty payload.
+      {},
+      namespace,
+      "discarded",
+      publicKey
+    );
+  }
+
+  /**
+   * Sends a Pioneer enrollment ping.
+   *
+   * The `creationDate` provided by the telemetry APIs will be used as the
+   * timestamp for considering the user enrolled in pioneer and/or the study.
+   *
+   * @param {String} [studyAddonid=undefined]
+   *        optional study id. It's sent in the ping, if present, to signal
+   *        that user enroled in the study.
+   */
+  async sendEnrollmentPing(studyAddonId) {
+    // If we were provided with a study id, then this is an enrollment to a study.
+    // Send the id alongside with the data and change the schema namespace to simplify
+    // the work on the ingestion pipeline.
+    if (typeof studyAddonId != "undefined") {
```

Comment:
```suggestion
    if (studyAddonId !== undefined) {
```

Reply:
Gah! Sorry about that, you pointed it out already in another PR.